cmd/tink-cli/cmd/template/create.go

```diff
@@ -22,54 +22,88 @@ var (
 
 // createCmd represents the create subcommand for template command
 var createCmd = &cobra.Command{
-	Use:     "create",
-	Short:   "create a workflow template ",
-	Example: "tink template create [flags]",
+	Use:   "create",
+	Short: "create a workflow template ",
+	Example: `tink template create [flags]
+cat /tmp/example.tmpl | tink template create -n example`,
+	PreRunE: func(c *cobra.Command, args []string) error {
+		if !isInputFromPipe() {
+			path, _ := c.Flags().GetString(fPath)
+			if path == "" {
+				return fmt.Errorf("either pipe the template or provide the required '--path' flag")
+			}
+		}
+		return nil
+	},
 	Run: func(c *cobra.Command, args []string) {
-		validateTemplate()
-		createTemplate(c, args)
+		var data []byte
+		if isInputFromPipe() {
+			data = readDataFromStdin()
+		} else {
+			data = readTemplateData()
+		}
+		if data != nil {
+			if err := tryParseTemplate(data); err != nil {
+				log.Println(err)
+				return
+			}
+			createTemplate(c, data)
+		}
 	},
 }
 
 func addFlags() {
 	flags := createCmd.PersistentFlags()
 	flags.StringVarP(&filePath, "path", "p", "", "path to the template file")
 	flags.StringVarP(&templateName, "name", "n", "", "unique name for the template (alphanumeric)")
-
-	createCmd.MarkPersistentFlagRequired(fPath)
 	createCmd.MarkPersistentFlagRequired(fName)
 }
 
-func validateTemplate() {
-	_, err := tt.ParseFiles(filePath)
-	if err != nil {
-		log.Fatalln(err)
+func tryParseTemplate(data []byte) error {
+	tmpl := *tt.New("")
+	if _, err := tmpl.Parse(string(data)); err != nil {
+		return err
 	}
+	return nil
 }
 
 func readTemplateData() []byte {
 	f, err := os.Open(filePath)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
 	}
 	defer f.Close()
 
 	data, err := ioutil.ReadAll(f)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
 	}
 	return data
 }
 
-func createTemplate(c *cobra.Command, args []string) {
-	req := template.WorkflowTemplate{Name: templateName, Data: readTemplateData()}
+func createTemplate(c *cobra.Command, data []byte) {
```

Comment:
While we're at it, I think we should reduce the globals used in this function and pass `templateName` as a parameter to the function to reduce spooky action. Could we do the same for `filePath` in `readTemplateData`?

Reply:
AFIK, since `templateName` and `filePath` are value receivers for the flags, they have to be global. But if there is a better to do that, please do let me know.